### DISCUSSION TITLE
ESP32 USB Port for PC

### DISCUSSION THREAD

**Discussion by elmariolo** [darthcloud/BlueRetro#447] [Ideas]
Is there any possibility to implement a "software usb" out for ESP32 to connect to the PC and emulate 2 or more HID controls?

Or maybe add an option to enable to send all the information of connected bluetooth devices throug the debug serial port and then we can create a driver for windows to interpret all the data as windows joysticks.

That would greatly open up the possibilities of playing on PC, since, for example, for Wiimotes and many other controls, drivers or scripts are required in order to work. (not to mention the tediousness of setting up bluetooth).

**Comment by kleffa** [comment]
I was thinking about, but it need a configurable usb serial, that most esp32 doesn't have.... but maybe you can search for a PS2-pc adapter and make some soledering
